src/async_wrap-inl.h

```diff
@@ -26,10 +26,21 @@
 
 #include "async_wrap.h"
 #include "base_object-inl.h"
+#include "node_errors.h"
 #include "node_internals.h"
+#include "v8.h"
 
 namespace node {
 
+using v8::Function;
+using v8::Isolate;
+using v8::Local;
+using v8::MaybeLocal;
+using v8::Name;
+using v8::String;
+using v8::Symbol;
+using v8::Value;
```

Comment:
We don’t add `using` statements to headers (we should probably have a linter rule against that, at least for the v8 symbols…)

Reply:
@addaleax, `grep "using v8::" src/*.h`

I am seeing this pattern in few files already. If I remove `using` phrase, is there any way I can elevate ?

Reply:
Yes, these other files should also not use it.

There are no shorthands in that case, we do spell out the entire type in headers.

Reply:
@addaleax,thanks. Reverted the changes. What is the rationale for this ?

Reply:
I mean, why we want to skip for headers ? Is there any performance reason or something else ?

Reply:
@HarshithaKP Because then the symbols are available in any source file that includes this header, even if that’s not obvious. For example, this code would fail to compile if `example.h` has a `using v8::Array;` line:

```c++
#include "example.h"

class Array {};
```